lib/Controller/ChatController.php

```diff
@@ -432,6 +436,22 @@ public function receiveMessages(int $lookIntoFuture,
 			return $response;
 		}
 
+		// Quickly scan messages for share IDs
+		$share_ids = [];
+		foreach ($comments as $comment) {
+			$verb = $comment->getVerb();
+			$message = $comment->getMessage();
+			if ($verb === 'object_shared') {
+				$data = json_decode($message, true);
+				$parameters = $data['parameters'];
+				$share_ids[] = $parameters['share'];
```

Comment:
Need to check with isset, so other shares like deck cards, locations, etc don't break the response.

```suggestion
		$shareIds = [];
		foreach ($comments as $comment) {
			$verb = $comment->getVerb();
			if ($verb === 'object_shared') {
				$message = $comment->getMessage();
				$data = json_decode($message, true);
				if (isset($data['parameters']['share'])) {
					$shareIds[] = $data['parameters']['share'];
				}
```